scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RemoveScala3OptionalBraces.scala

```diff
@@ -0,0 +1,133 @@
+package org.scalafmt.rewrite
+
+import scala.meta.Defn
+import scala.meta.Template
+import scala.meta.Term
+import scala.meta.tokens.Token
+
+import org.scalafmt.config.RewriteScala3Settings
+import org.scalafmt.config.ScalafmtConfig
+import org.scalafmt.internal.FormatToken
+import org.scalafmt.internal.FormatTokens
+import org.scalafmt.util.TreeOps
+
+object RemoveScala3OptionalBraces extends FormatTokensRewrite.RuleFactory {
+
+  override def enabled(implicit style: ScalafmtConfig): Boolean =
+    style.runner.dialect.allowSignificantIndentation &&
+      style.rewrite.scala3.removeOptionalBraces
+        .ne(RewriteScala3Settings.RemoveOptionalBraces.no)
+
+  override def create(ftoks: FormatTokens): FormatTokensRewrite.Rule =
+    new RemoveScala3OptionalBraces(ftoks)
+
+}
+
+private class RemoveScala3OptionalBraces(ftoks: FormatTokens)
+    extends FormatTokensRewrite.Rule {
+
+  import FormatTokensRewrite._
+
+  private def allowOldSyntax(implicit style: ScalafmtConfig): Boolean =
+    ConvertToNewScala3Syntax.enabled ||
+      style.rewrite.scala3.removeOptionalBraces
+        .eq(RewriteScala3Settings.RemoveOptionalBraces.oldSyntaxToo)
+
+  override def enabled(implicit style: ScalafmtConfig): Boolean =
+    RemoveScala3OptionalBraces.enabled
+
+  override def onToken(implicit
+      ft: FormatToken,
+      style: ScalafmtConfig
+  ): Option[Replacement] = Option {
+    ft.right match {
+      case x: Token.LeftBrace // skip empty brace pairs
+          if !ftoks.nextNonComment(ftoks.next(ft)).right.is[Token.RightBrace] =>
+        ft.meta.rightOwner match {
+          case t: Term.Block if t.stats.nonEmpty => onLeftForBlock(t)
+          case t: Template if t.stats.nonEmpty || t.self.tokens.nonEmpty =>
+            if (t.parent.exists(_.is[Defn.Given])) removeToken
+            else
+              replaceToken(":")(new Token.Colon(x.input, x.dialect, x.start))
+          case _: Term.For if allowOldSyntax || {
+                val rbFt = ftoks(ftoks.matching(ft.right))
+                ftoks.nextNonComment(rbFt).right.is[Token.KwDo]
+              } =>
+            removeToken
+          case _: Term.ForYield => removeToken
+          case _: Term.Match => removeToken
+          case _: Term.Try => removeToken
+          case _: meta.Ctor.Secondary
+              if ftoks.prevNonComment(ft).left.is[Token.Equals] =>
```

Comment:
for now, simply making sure it's not procedure syntax, since it's still occasionally supported. if merge scalameta/scalameta#2313 and release a new version, will add the new flag here.

Reply:
Sure, however I think we will never actually have a use for this. We need the Scala 3 dialect to use this rewrite and any codebase that uses Scala 3 will not be compiling with the procedure syntax. So once the new version of scalameta is released we will just remove this (or at least the tests).

Reply:
of course!